block/genhd.c

```diff
@@ -939,6 +940,8 @@ void del_gendisk(struct gendisk *disk)
 	if (!sysfs_deprecated)
 		sysfs_remove_link(block_depr, dev_name(disk_to_dev(disk)));
 	pm_runtime_set_memalloc_noio(disk_to_dev(disk), false);
+	io_filter_bpf_free(disk);
```

Comment:
If we do as I mentioned above, we'll need to swap the order of mutex_destroy() and io_filter_bpf_free() here.

Reply:
done